### PR TITLE
checker: disallow indexing `mut struct`, when using as parameters

### PR DESCRIPTION
**Pull request by shove70**
1. make an error for indexing the `mut struct`
2. Add tests.

Currently, `zz &MyStruct` and `zz[0]` V give a warning:
```
warning: pointer indexing is only allowed in `unsafe` blocks
```
but `mut zz MyStruct` and `zz[0]` do not

### CODE REVIEW

File: vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out [hunk 1]
```diff
@@ -0,0 +1,34 @@
+vlib/v/checker/tests/index_sumtype_interface_struct_params_err.vv:11:9: warning: pointer indexing is only allowed in `unsafe` blocks
+    9 |     _ = bar[0]
+   10 | 
+   11 |     _ = st1[0]
+      |            ~~~
```
Comment (spytheman) [inline vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out:5]: This should be an error, not a notice. `mut st1 Struct` does not allow indexing *at all*, even in unsafe.

File: vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+   12 |     _ = st2[0]
+   13 |     _ = st3[0]
+vlib/v/checker/tests/index_sumtype_interface_struct_params_err.vv:12:9: warning: pointer indexing is only allowed in `unsafe` blocks
+   10 | 
+   11 |     _ = st1[0]
+   12 |     _ = st2[0]
+      |            ~~~
```
Comment (spytheman) [inline vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out:12]: that is good, `st2 &Struct` should allow indexing within unsafe{}

File: vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    8 |     _ = foo[0]
+    9 |     _ = bar[0]
+      |            ~~~
+   10 | 
+   11 |     _ = st1[0]
+vlib/v/checker/tests/index_sumtype_interface_struct_params_err.vv:13:9: error: type `Struct` does not support indexing
+   11 |     _ = st1[0]
+   12 |     _ = st2[0]
+   13 |     _ = st3[0]
+      |            ~~~
```
Comment (spytheman) [inline vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out:33]: this is good, `st3 Struct` can not be indexed, regardless of unsafe, so it should be an error, as it is.

File: vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out (hunk 1 above)
Comment (spytheman) [reply vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out:5]: It should be an error, for the same reason that `st3[0]` is.

**Comment by Delta456** [comment]
Is this PR co-related to https://github.com/vlang/v/pull/19949?

File: vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out (hunk 1 above)
Comment (shove70) [reply vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out:5]: Oh, I see! Thanks!  👍

**Comment by spytheman** [comment]
> Is this PR co-related to #19949?

afaik no

File: vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out (hunk 1 above)
Comment (shove70) [reply vlib/v/checker/tests/index_sumtype_interface_struct_params_err.out]: done, thanks for review!

**Review by spytheman — APPROVED** [review]

**Pull request merged**